File: code/TicTactoe.py
import random


def display_board(board):
	'''
	displat board
	'''
	print("| 7 | 8 | 9 |   | {} | {} | {} |".format(board[7],board[8],board[9]))
	print("-------------   -------------")
	print("| 4 | 5 | 6 |   | {} | {} | {} |".format(board[4],board[5],board[6]))
	print("-------------   -------------")
	print("| 1 | 2 | 3 |   | {} | {} | {} |".format(board[1],board[2],board[3]))

def player_input():
	mark=input("Player 1: What do you want choose X or O ?")
	mark=mark.upper()
	while mark not in ('X','O'):
		mark=input("Player 1: What do you want choose X or O ?")
		mark=mark.upper()

	if mark == 'X':
		return ('X','O')
	else:
		return ('O','X')

def place_marker(board,marker,position):
	board[position]=marker

def win(board,mark):

	return board[1]==board[2]==board[3]==mark or board[4]==board[5]==board[6]==mark or board[7]==board[8]==board[9]==mark or \
		board[1]==board[4]==board[7]==mark or board[2]==board[5]==board[8]==mark or board[3]==board[6]==board[9]==mark or \
		board[1]==board[5]==board[9]==mark or board[3]==board[5]==board[7]==mark
		
def who_first():
	'''
		which player will start
		if return True player 1
		else player 2
	'''
	x=random.randint(0,1)
	return x%2==0

def free_def(board):
	return ' ' in board

def next_pos(board):

	mov=int(input("Give me num between 1 to 9: "))
	while True:
		if mov in (1,2,3,4,5,6,7,8,9) and ' ' in board[mov] :
			return mov
		else:	
			mov=int(input("Give me num between 1 to 9: "))



#main
print("Tic Tac Toe Game by Orel Gershonovich")
b=[' ']*10
b[0]=1
player1,player2=player_input()
print("player 1 is: {}\nplayer 2 is: {}\n".format(player1,player2))
display_board(b)
p1=who_first()

if p1==True:
	while True:	
		print("player 1 play")
		player1Mov=next_pos(b)
		place_marker(b,player1,player1Mov)
		display_board(b)
		space=free_def(b)
		if space == True:
			pass
		else:
			break

		res=win(b,player1)
		if  res == True:
			print("player 1 win!!!")
			break
			
		print("player 2 play")
		player2Mov=next_pos(b)
		place_marker(b,player2,player2Mov)
		display_board(b)
		space=free_def(b)
		if space == True:
			pass
		else:
			break

		res=win(b,player2)
		if  res == True:
			print("player 2 win!!!")
			break


else:
	while True:
		print("player 2 play")
		player2Mov=next_pos(b)
		place_marker(b,player2,player2Mov)
		display_board(b)
		space=free_def(b)
		if space == True:
			pass
		else:
			break

		res=win(b,player2)
		if  res == True:
			print("player 2 win!!!")
			break

		print("player 1 play")
		player1Mov=next_pos(b)
		place_marker(b,player1,player1Mov)
		display_board(b)
		space=free_def(b)
		if space == True:
			pass
		else:
			break

		res=win(b,player1)
		if  res == True:
			print("player 1 win!!!")	
			break

print("End!!!")

File: code/TicTacToe.cpp
#include <iostream>
using namespace std;

void displayMap(char map[][3],int size){
	int i=0,j=0,z=0;

	cout << "   0   1   2"<<endl;
	for(i=0;i<size;i++){
		cout<<z;
		z++;
		for(j=0;j<size;j++){
			cout<<" "<<"|"<<map[i][j]<<"|";
		}
	cout<<endl;		
	}
	z=0;
}

bool chaeckMapPlayer1(char map[][3],int size,int line,int row)
{
	int win=0;
	for(int i=0;i<3;i++){
		if(map[line][i]=='X'){
			win++;				
		}
		else{
			win=0;
			break;
		}
	
		if(win==3){
			cout<<"player 1 win"<<endl;
			return true;
		}

	}
	
	for(int i=0;i<3;i++){
		if(map[i][row]=='X'){
			win++;				
		}
		else{
			win=0;
			break;
		}
	
		if(win==3){
			cout<<"player 1 win"<<endl;
			return true;
		}
	}

	if(map[0][0] == 'X' && map[1][1] == 'X' && map[2][2] == 'X'){
		cout<<"player 1 win"<<endl;
			return true;
	}

	if(map[0][2] == 'X' && map[1][1] == 'X' && map[2][0] == 'X'){
		cout<<"player 1 win"<<endl;
			return true;
	}

	return false;
}

bool chaeckMapPlayer2(char map[][3],int size,int line,int row)
{
	int win=0;
	for(int i=0;i<3;i++){
		if(map[line][i]=='O'){
			win++;				
		}
		else{
			win=0;
			break;
		}
	
		if(win==3){
			cout<<"player 2 win"<<endl;
			return true;
		}

	}
	
	for(int i=0;i<3;i++){
		if(map[i][row]=='O'){
			win++;				
		}
		else{
			win=0;
			break;
		}
	
		if(win==3){
			cout<<"player 2 win"<<endl;
			return true;
		}

	}

	if(map[0][0] == 'O' && map[1][1] == 'O' && map[2][2] == 'O'){
		cout<<"player 2 win"<<endl;
			return true;
	}

	if(map[0][2] == 'O' && map[1][1] == 'O' && map[2][0] == 'O'){
		cout<<"player 2 win"<<endl;
			return true;
	}
	return false;
}

bool gameOvaer(char map[][3],int size)
{
	for(int i=0;i<size;i++){
		for(int j=0;j<size;j++){
			if(map[i][j]=='A'){
				return false;
			}
		}
	}
	return true;
}




int main()
{
	int line,row;
	char game1,game2;
	char map[3][3]={'A','A','A','A','A','A','A','A','A'};
	bool opp=false,gameover=false;
	cout<<"what do you want to choose: X or O"<<endl;
	cin>>game1;

	while(('X' != game1) && ('O' != game1)){
		cout<<"you can choose X or O"<<endl;
		cin>>game1;
	}

	cout<<"OK lets start:\n\n"<<endl;
	cout<<"Player 1 is X \nplayer 2 is O\n"<<endl;
	cout<<"player 1 starts to play"<<endl;
	displayMap(map,3);

	while(opp == false)
	{
		cout<<"player1: choose where do you want to put X in the board line and row"<<endl;
		cin >> line;
		cin >> row;
		map[line][row]='X';
		displayMap(map,3);
		opp=chaeckMapPlayer1(map,3,line,row);
		gameover=gameOvaer(map,3);
		if(opp == true)
			break;
		if(gameover == true)
		{
			cout<<"Game Over"<<endl;
			break;
		}
		
		cout<<"player2: choose where do you want to put X in the board line and row"<<endl;
		cin >> line;
		cin >> row;
		map[line][row]='O';
		displayMap(map,3);
		opp=chaeckMapPlayer2(map,3,line,row);
		gameover=gameOvaer(map,3);
		if(opp == true)
			break;
		if(gameover == true)
		{
			cout<<"Game Over"<<endl;
			break;
		}
	}

	cout<<"good bye"<<endl;
	
	return 0;
}